Fix methods with no arguments

pablomayobre opened the pull request:
Methods with no arguments should still have an arguments (`args`) table which only has `self` in it. Fixes #16

Renamed `includeDummy` to `isMethod` in the process

Note: The second commit is optional

rm-code commented:
```
bash update.sh
Cloning into 'api'...                                                                                                                                                   
remote: Counting objects: 156, done.                                                                                                                                    
remote: Compressing objects: 100% (141/141), done.                                                                                                                      
remote: Total 156 (delta 74), reused 56 (delta 5), pack-reused 0                                                                                                        
Receiving objects: 100% (156/156), 119.46 KiB | 48.00 KiB/s, done.                                                                                                      
Resolving deltas: 100% (74/74), done.                                                                                                                                   
Generating LOVE snippets ...                                                                                                                                            
Found API file for LÖVE version 0.10.2                                                                                                                                  
Generating functions ...                                                                                                                                                
lua: json-generator.lua:44: bad argument #1 to 'ipairs' (table expected, got nil)                                                                                       
stack traceback:                                                                                                                                                        
        [C]: in function 'ipairs'                                                                                                                                       
        json-generator.lua:44: in function 'createArguments'                                                                                                            
        json-generator.lua:80: in function 'createVariant'                                                                                                              
        json-generator.lua:101: in function 'createFunction'                                                                                                            
        json-generator.lua:123: in function 'createLoveFunctions'                                                                                                       
        json-generator.lua:151: in function 'createGlobals'                                                                                                             
        json-generator.lua:223: in function 'createJSON'                                                                                                                
        json-generator.lua:237: in main chunk                                                                                                                           
        [C]: in ?                                
```

pablomayobre commented:
Oh ipairs, I thought it was a numerical loop haha I'll fix that

pablomayobre committed:
Fix methods with no arguments

Methods with no arguments should still have an argument table which only has self included. Fixes #16.

Renamed includeDummy to isMethod in the process

pablomayobre committed:
Remove unnecessary if checks

createArguments doesn't need to check if the description is nil

Also having an empty args tables for functions that don't take arguments is more explicit than having no args table at all
